draft-ietf-tls-extended-key-update.md

```diff
@@ -780,19 +780,22 @@ need to use the newly derived exporter secret to generate Exported Keying Materi
 used as the "Secret" in the exporter function, defined in
 {{Section 7.5 of TLS}}, to generate EKM, ensuring that
 the exported keying material is aligned with the updated security context.
+The newly derived exporter secret is cryptographically independent of
+previous exporter secrets.
 
 When a new exporter secret becomes active following a successful Extended
 Key Update, the TLS or DTLS implementation would have to provide an
 asynchronous notification to the application indicating that:
 
-* A new epoch has become active; and
+* A new epoch has become active, and the implementation can
+  include the corresponding epoch identifier. Applications receiving an
+  epoch identifier can use it to request keying material for that
+  specific epoch through an epoch-aware exporter interface. The epoch becomes
+  available for export once both peers have completed the Extended Key Update exchange.
+  In TLS, this identifier represents a local logical counter that may differ between peers.
```

Comment:
This is racy. In order to prevent race conditions, the notification must be delivered only after both sides have activated the new epoch for epoch-aware export. This implies that the new epoch must be activated before the EKU exchange actually completes.

It is critical for interoperability that both sides agree exactly when the new epoch is activated, and exactly when the new epoch is notified.

Reply:
Fixed, please check.